File: CS/BindingToLegacyViewModels/DataBindingViaMVVMContextUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSampleBindingToLegacyViewModels {
    public partial class DataBindingViaMVVMContextUserControl : XtraUserControl {
        public DataBindingViaMVVMContextUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            TextEdit editor = new TextEdit();
            editor.Dock = DockStyle.Top;
            editor.Properties.NullValuePrompt = "Please, enter the Title here...";
            editor.Properties.NullValuePromptShowForEmptyValue = true;
        
            SimpleButton commandButton = new SimpleButton();
            commandButton.Dock = DockStyle.Top;
            commandButton.Text = "Report the Title property value";
        
            commandButton.Parent = this;
            editor.Parent = this;
            #endregion SetUp

            #region #dataBindingViaMVVMContext
            var legacyViewModel = new LegacyViewModel("Legacy ViewModel");
            // initialize the MVVMContext with the specific ViewModel's instance
            mvvmContext.SetViewModel(typeof(LegacyViewModel), legacyViewModel);
            // Data binding for the Title property (via MVVMContext API)
            mvvmContext.SetBinding(editor, e => e.EditValue, "Title");
            // UI binding for the Report command
            commandButton.Click += (s, e) => XtraMessageBox.Show(legacyViewModel.Title);
            #endregion #dataBindingViaMVVMContext
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        public class LegacyViewModel {
            string titleCore;
            public LegacyViewModel(string title) {
                this.titleCore = title;
            }
            public virtual string Title {
                get { return titleCore; }
                set {
                    if(titleCore == value) return;
                    titleCore = value;
                    OnTitleChanged();
                }
            }
            void OnTitleChanged() {
                EventHandler h = TitleChanged;
                if(h != null) h(this, EventArgs.Empty);
            }
            public event EventHandler TitleChanged;
        }
    }
}

File: VB/BindingToLegacyViewModels/DataBindingViaMVVMContextUserControl.vb
Imports DevExpress.MVVM
Imports DevExpress.MVVM.DataAnnotations
Imports DevExpress.MVVM.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSampleBindingToLegacyViewModels
    Partial Public Class DataBindingViaMVVMContextUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim editor As New TextEdit()
            editor.Dock = DockStyle.Top
            editor.Properties.NullValuePrompt = "Please, enter the Title here..."
            editor.Properties.NullValuePromptShowForEmptyValue = True
        
            Dim commandButton As New SimpleButton()
            commandButton.Dock = DockStyle.Top
            commandButton.Text = "Report the Title property value"
        
            commandButton.Parent = Me
            editor.Parent = Me

            Dim legacyViewModel = New LegacyViewModel("Legacy ViewModel")
            ' initialize the MVVMContext with the specific ViewModel's instance
            mvvmContext.SetViewModel(GetType(LegacyViewModel), legacyViewModel)
            ' Data binding for the Title property (via MVVMContext API)
            mvvmContext.SetBinding(editor, Function(e) e.EditValue, "Title")
            ' UI binding for the Report command
            AddHandler commandButton.Click, Sub(s, e) XtraMessageBox.Show(legacyViewModel.Title)
        End Sub
         Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        Public Class LegacyViewModel
            Private titleCore As String
            Public Sub New(ByVal title As String)
                Me.titleCore = title
            End Sub
            Public Overridable Property Title() As String
                Get
                    Return titleCore
                End Get
                Set(ByVal value As String)
                    If titleCore = value Then
                        Return
                    End If
                    titleCore = value
                    OnTitleChanged()
                End Set
            End Property
            Private Sub OnTitleChanged()
                Dim h As EventHandler = TitleChangedEvent
                If h IsNot Nothing Then
                    h(Me, EventArgs.Empty)
                End If
            End Sub
            Public Event TitleChanged As EventHandler
        End Class
    	End Class
End Namespace